### ISSUE TITLE
Fix `function-url-quotes` false positives for SCSS variable and `@` character 

### ISSUE THREAD

**Issue by tavelli** [stylelint/stylelint#7404]
### What minimal example or steps are needed to reproduce the bug?

```scss

.class {
  background-image: url($asset-prefix + 'assets/images/fa_calculator_blue@2x.png');
}

.otherclass {
    background: url($asset-prefix + 'assets/images/HSA_Icons-31.png');
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard-scss"
}
```

### How did you run Stylelint?

```shell
"stylelint \"apps/**/*.scss\" \"libs/**/*.scss\"" --fix
```

### Which Stylelint-related dependencies are you using?

```json
"stylelint": "^16.0.2",
"stylelint-config-standard-scss": "^12.0.0",
 ```

### What did you expect to happen?

nothing should happen. format is valid/conforming.

```scss
.class {
  background-image: url($asset-prefix + 'assets/images/fa_calculator_blue@2x.png');
}

.otherclass {
    background: url($asset-prefix + 'assets/images/HSA_Icons-31.png');
}
```

### What actually happened?

added extraneous double quotes around entire thing 

```scss
.class {
  background-image: url("$asset-prefix + 'assets/images/fa_calculator_blue@2x.png'");
}

.otherclass {
    background: url($asset-prefix + 'assets/images/HSA_Icons-31.png');
}
```

### Do you have a proposal to fix the bug?

the `@` symbol in the image path seems to be the problem, when i remove that it does not add wrapping double quotes...

**Comment by jeddy3** [comment]
@tavelli Thanks for the report and for using the template.

We'll probably need to improve the heuristics in [`isStandardSyntaxUrl`](https://github.com/stylelint/stylelint/blob/main/lib/utils/isStandardSyntaxUrl.mjs).

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by mattxwang** [comment]
Ended up taking a stab at diagnosing the issue and wrote a longer explanation in #7416!
